src/sshagent/SSHAgent.cpp

```diff
@@ -201,9 +211,18 @@ bool SSHAgent::addIdentity(OpenSSHKey& key, quint32 lifetime, bool confirm)
         return false;
     }
 
+    OpenSSHKey keyCopy = key;
+    keyCopy.clearPrivate();
+    m_addedKeys[keyCopy] = removeOnLock;
```

Comment:
Previously, in 2.3.4, only those keys with the option "remove at lock" were removed from the ssh agent upon closing keepassxc.
see #2980